File: LeetCode题目集/1. Two Sum(easy).cpp
/*
 @Date    : 2018-01-05 13:18:10
 @Author  : 酸饺子 (changzheng300@foxmail.com)
 @Link    : https://github.com/SourDumplings
 @Version : $Id$
*/

/*
https://leetcode.com/problems/two-sum/description/
 */

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target)
    {
        for(int i = 0; i != nums.size(); ++i)
        {
            for(int j = i + 1; j != nums.size(); ++j)
            {
                if(nums[i] + nums[j] == target)
                    return *(new vector<int>{i, j});
            }
        }
    }
};

File: OJ practices/LeetCode题目集/1. Two Sum(easy).py
'''
@Autor: SourDumplings
@Date: 2020-01-24 09:23:24
@Link: https://github.com/SourDumplings/
@Email: changzheng300@foxmail.com
@Description: 
'''
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        l = len(nums)
        d = dict()
        for i, n in enumerate(nums):
            d[n] = i
        for i in range(l):
            r = target - nums[i]
            j = d.get(r)
            if j is not None and i != j:
                return [i, d[r]]
